dvc/remote/http.py

```diff
@@ -43,24 +28,26 @@ def __init__(self, repo, config):
         self.path_info = self.path_cls(url) if url else None
 
     def _download(self, from_info, to_file, name=None, no_progress_bar=False):
-        callback = None
-        if not no_progress_bar:
-            total = self._content_length(from_info.url)
-            if total:
-                callback = ProgressBarCallback(name, total)
-
         request = self._request("GET", from_info.url, stream=True)
-
-        with open(to_file, "wb") as fd:
-            transferred_bytes = 0
-
-            for chunk in request.iter_content(chunk_size=self.CHUNK_SIZE):
-                fd.write(chunk)
-                fd.flush()
-                transferred_bytes += len(chunk)
-
-                if callback:
-                    callback(transferred_bytes)
+        total = self._content_length(from_info.url)
```

Comment:
No need to query for content length in `no_progress_bar is True`. 

Actually we should never query for it, this is an inefficiency in our code. `total` is `request.headers.get('Content-Length')`. And `_content_length()` method should be dropped.

Reply:
yes I'd favour using `request.headers.get('Content-Length')` without checking `no_progress_bar` since it's a negligible overhead